Python/ceval.c

```diff
@@ -4637,40 +4643,48 @@ _PyEval_EvalFrameDefault(PyThreadState *tstate, InterpreterFrame *frame, int thr
 
             // Check if the call can be inlined or not
             PyObject *function = PEEK(oparg + 1);
-            int inline_call = 0;
             if (Py_TYPE(function) == &PyFunction_Type) {
                 PyCodeObject *code = (PyCodeObject*)PyFunction_GET_CODE(function);
-                int is_coro = code->co_flags & (CO_GENERATOR | CO_COROUTINE | CO_ASYNC_GENERATOR);
-                inline_call = (is_coro || cframe.use_tracing) ? 0 : 1;
+                STACK_SHRINK(oparg + 1);
+                if (code->co_flags & (CO_GENERATOR | CO_COROUTINE | CO_ASYNC_GENERATOR)) {
+                    PyObject *locals = code->co_flags & CO_OPTIMIZED ? NULL : PyFunction_GET_GLOBALS(function);
+                    res = make_coro(
+                        tstate, PyFunction_AS_FRAME_CONSTRUCTOR(function), locals, stack_pointer+1, oparg, NULL);
+                    for (int i = 0; i < oparg+1; i++) {
+                        Py_DECREF(stack_pointer[i]);
+                    }
+                    PUSH(res);
+                    if (res == NULL) {
+                        goto error;
+                    }
```

Comment:
The control flow here is quite confusing. This is falling over to the `DISPATH()` outside the conditional, which is quite far from here.

What's the advantage of doing this here? Is this because you plan to also inline coroutines?

Reply:
This PR already inlines coroutine creation. I've moved the code around a bit to make it clearer.